### PR TITLE
Update Shell packages to use TaskListItem instead of Task class name

### PR DESCRIPTION
**Pull request by dmonroym**
`Microsoft.VisualStudio.Shell.Task` has been renamed to `Microsoft.VisualStudio.Shell.TaskListItem`. This change updates the versions of the packages to take this new change and we then had to change the use on our code in 2 files so it wouldn't use `System.Threading.Tasks.Task` instead.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/7255)

### CODE REVIEW

File: build/import/Packages.targets [hunk 1]
```diff
@@ -44,34 +44,34 @@
...
     <PackageReference Update="Microsoft.VisualStudio.Composition"                                     Version="16.9.20"/>
     <PackageReference Update="Microsoft.VisualStudio.Data.Core"                                       Version="9.0.21023" />
     <PackageReference Update="Microsoft.VisualStudio.DataDesign.Common"                               Version="17.0.0-preview-2-31223-026"/>
     <PackageReference Update="Microsoft.VisualStudio.Data.Services"                                   Version="9.0.21023" />
     <PackageReference Update="Microsoft.VisualStudio.DataTools.Interop"                               Version="17.0.0-preview-2-31223-026" />
     <PackageReference Update="Microsoft.VisualStudio.Designer.Interfaces"                             Version="17.0.0-preview-2-31223-026" />
-    <PackageReference Update="Microsoft.VisualStudio.ImageCatalog"                                    Version="17.0.0-preview-2-31223-026" />
+    <PackageReference Update="Microsoft.VisualStudio.ImageCatalog"                                    Version="17.0.0-previews-1-31321-016" />
```
Comment (swesonga) [inline build/import/Packages.targets:54]: strange that this is now versioned as "previews"

Comment (dmonroym) [reply build/import/Packages.targets:54]: Yes, not sure why they switched from preview-2 to previews-1 again but this is a higher version than what was there before

File: build/import/Packages.targets [hunk 2]
```diff
@@ -44,34 +44,34 @@
...
+    <PackageReference Update="Microsoft.VisualStudio.Telemetry"                                       Version="16.3.176" />
     <PackageReference Update="Microsoft.VisualStudio.Settings.15.0"                                   Version="16.8.30406.155-pre" />
     <PackageReference Update="Microsoft.VisualStudio.Setup.Configuration.Interop"                     Version="2.3.2262-g94fae01e" />
     <PackageReference Update="Microsoft.VisualStudio.SDK.EmbedInteropTypes"                           Version="15.0.34" />
-    <PackageReference Update="Microsoft.VisualStudio.Shell.15.0"                                      Version="17.0.0-preview-2-31223-026" />
-    <PackageReference Update="Microsoft.VisualStudio.Shell.Design"                                    Version="17.0.0-preview-2-31223-026" />
-    <PackageReference Update="Microsoft.VisualStudio.Shell.Framework"                                 Version="17.0.0-preview-2-31223-026" />
+    <PackageReference Update="Microsoft.VisualStudio.Shell.15.0"                                      Version="17.0.0-previews-1-31321-016" />
```
Comment (swesonga) [inline build/import/Packages.targets:61]: Why the downgrade from preview-2 to preview 1?

**Review by swesonga — APPROVED** [review]

File: build/import/Packages.targets (hunk 2 above)
Comment (dmonroym) [reply build/import/Packages.targets:61]: previews-1 is an upgrade from preview-2. Strange as the numbers don't make sense but you can confirm by looking at the publish dates of packages here: https://dev.azure.com/azure-public/vside/_packaging?_a=feed&feed=vssdk%40Local

**Review by jjmew — APPROVED** [review]
:shipit:

**Review by tmeschter — APPROVED** [review]

**Review by MiYanni — APPROVED** [review]

**Comment by dmonroym** [comment]
Opens #7256

**Pull request merged**